FinSearchUnified/Bundle/SearchBundleFindologic/SortingHandler/PopularitySortingHandler.php

```diff
@@ -26,10 +26,10 @@ public function supportsSorting(SortingInterface $sorting)
      * Handles the passed sorting object.
      *
      * @param SortingInterface $sorting
-     * @param QueryBuilder $query
+     * @param NewQueryBuilder $query
      * @param ShopContextInterface $context
      */
-    public function generateSorting(SortingInterface $sorting, QueryBuilder $query, ShopContextInterface $context)
+    public function generateSorting(SortingInterface $sorting, NewQueryBuilder $query, ShopContextInterface $context)
```

Comment:
I'd suggest using the interface here.

Reply:
We do not have any interface for this. We have an abstract class, which is being used here. (Applicable to other comments below as well)